core/sync.js

```diff
@@ -307,6 +307,10 @@ class Sync {
 
     if (metadata.shouldIgnore(doc, this.ignore)) {
       return this.pouch.setLocalSeqAsync(change.seq)
+    } else if (!metadata.wasSynced(doc) && doc.deleted) {
+      doc._deleted = true
```

Comment:
[nit] I think we can even do `doc = { _deleted: true }` to not keep the other fields (changes feed)

Reply:
We still need the `_id`. Do you think we'd get some performance or storage gain from this change?

Reply:
It would take a bit less storage. But the main motivation should be to avoid getting stale data (for example, it may be used to know that the file has been removed remotely during the migration phase).

Reply:
Noted. To make the difference between documents `_deleted` during the migration phase and those "erased" after we've synchronized the deletion, I was thinking of using the sides. If they're up-to-date then it means there's nothing to do (except on that specific line where I check for both `_deleted` and `deleted` now).

Reply:
Ah, I can't do what I proposed, because moves are still erasing source documents.
I'll try your idea.